front/src/modules/trainschedule/components/Timetable/Timetable.tsx

```diff
@@ -78,20 +69,60 @@ export default function Timetable({
   const [deleteTrainScheduleById] = osrdEditoastApi.endpoints.deleteTrainScheduleById.useMutation();
   const [deleteTrainSchedules] = osrdEditoastApi.endpoints.deleteTrainSchedule.useMutation();
 
-  const [getTimetableWithTrainSchedulesDetails] = osrdEditoastApi.useLazyGetTimetableByIdQuery();
-  const [getTimetableConflicts] = osrdEditoastApi.useLazyGetTimetableByIdConflictsQuery();
+  const { data: timetable, refetch: refetchTimetable } =
+    osrdEditoastApi.endpoints.getTimetableById.useQuery(
+      { id: timetableID as number },
+      { skip: !timetableID || infraState !== 'CACHED' }
+    );
+
+  const { data: conflicts = [], refetch: refetchConflicts } =
+    osrdEditoastApi.endpoints.getTimetableByIdConflicts.useQuery(
+      { id: timetableID as number },
+      { skip: !timetableID }
+    );
+
+  useEffect(() => {
+    if (timetable) {
+      getSimulationResults(timetable);
+      refetchConflicts();
+    }
+  }, [timetable]);
+
+  const trainsDurationsIntervals = useMemo(
+    () =>
+      timetable?.train_schedule_summaries
+        ? findTrainsDurationsIntervals(timetable.train_schedule_summaries)
+        : [],
+    [timetable]
+  );
+
+  const isTrainFiltered = (train: ScheduledTrain, debounce?: string): boolean => {
+    if (debounce) {
+      return (
+        !train.train_name.toLowerCase().includes(debouncedTerm.toLowerCase()) &&
+        !train.labels.join('').toLowerCase().includes(debouncedTerm.toLowerCase())
+      );
+    }
+    return false;
+  };
```

Comment:
Could this be done like that or the readability is bad ?
```suggestion
  const isTrainFiltered = (train: ScheduledTrain, debounce?: string): boolean => 
  !!debounce 
  && !train.train_name.toLowerCase().includes(debouncedTerm.toLowerCase()) 
  && !train.labels.join('').toLowerCase().includes(debouncedTerm.toLowerCase());
```

Reply:
Argh, I think it not very readable :/ I would rather keep it like this, if you are fine with it